test/functional/android/App.spec.js

```diff
@@ -72,21 +73,34 @@ describe("App", () => {
     });
   });
 
-  describe('Cryptographic Functionality', () => {
-    it('calculates & displays hmac of latlng when location changes', () => {
-
+  describe('Cryptographic Functionality', function(){
+    this.timeout(100000);
```

Comment:
that's a long timeout! 

again, not for this PR, but probably worth keeping our eye on this. do we think it's the crypto that's making it take this long? should we look into profilers to figure out for sure?

as a guess (just a guess until we profile): perhaps the re-hashing 1000 times specified by the `iv` field? if so, perhaps we could write some benchmarking unit tests with increasingvalues for the `iv` field, graph the output and try to get a sense of the impact re-hashing N times has on performance?